Student Name: Eliana Kariuki
Student Pace: DSF-FT12-Hybrid
Instructor Name: Samuel Karu

In [16]:
import pandas as pd
import numpy as np
import scipy as stats
import matplotlib.pyplot as plt
import seaborn as sns

In [17]:
movie_df = pd.read_csv(r'C:\Users\HomePC\Documents\labs\Statistics\box-office-movie-insights\zipped-data\bom.movie_gross.csv.gz')
movie_df.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [18]:
# Display the first few rows of the DataFrame  
movie_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [19]:
# Display the column names of the DataFrame
movie_df.columns

Index(['title', 'studio', 'domestic_gross', 'foreign_gross', 'year'], dtype='object')

In [20]:
# Display the shape of the DataFrame
movie_df.shape

(3387, 5)

In [21]:
# Display the number of missing values in each column
movie_df.isnull().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

In [ ]:
# Display the statistical summary information of the DataFrame
movie_df.describe() 

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

The dataset contains information about box office movie insights. It includes 3,387 rows and 5 columns, with each row representing a movie. The columns provide the following details:

1. `title`: The title of the movie.
2. `studio`: The studio that produced the movie.
3. `domestic_gross`: The domestic box office gross revenue (in dollars).
4. `foreign_gross`: The foreign box office gross revenue (in dollars).
5. `year`: The year the movie was released.

The dataset has some missing values, particularly in the `studio`, `domestic_gross`, and `foreign_gross` columns. It can be used to analyze trends in movie revenues, compare domestic and foreign earnings, and explore the performance of different studios over time.